### PR TITLE
feat: ```greekLetter()```

### LINKED ISSUE #558: Random greek alphabet letter
### Clear and concise description of the problem

As a user I want to generate a random greek letter so that I can prepend this to a name of an object.

I would like to have the option to get a short or a long version of a letter (`α` or `'alpha'`).
Maybe `Α` and `άλφα` could also somehow get returned.

### Suggested solution

```ts
const cameraName = 'Camera ' + faker.<?>.greekLetter({ format: 'long' });

const cameraName2 = faker.helpers.mustache('Camera {{greekLetter({format:"long"})}}');
```

I'm open to other ideas

### Alternative

_No response_

### Additional context

https://en.wikipedia.org/wiki/Greek_alphabet#Letters

### PR DESCRIPTION
**Pull request by import-brain**
fixes #558

How would I write tests for this? I was planning on using Vitest's ```toContain``` method, but according to the Vitest docs, toContain only accepts a string parameter, and I was planning on feeding an array of Greek letters to it. My current method is a somewhat workaround, but it's super janky. Any ideas?

### CODE REVIEW

File: src/random.ts [hunk 1]
```diff
@@ -557,4 +557,78 @@ export class Random {
...
+      'chi',
+      'psi',
+      'omega',
+    ];
+
+    if (options.format === 'short') {
+      return this.faker.random.arrayElement(shortGreekLetters);
+    } else if (options.format === 'long') {
+      return this.faker.random.arrayElement(longGreekLetters);
+    }
```
Comment (xDivisionByZerox) [inline src/random.ts:632]: Missing default case. Currently, your function could return undefined.

**Comment by xDivisionByZerox** [comment]
> How would I write tests for this? I was planning on using Vitest's toContain method, but according to the Vitest docs, toContain only accepts a string parameter, and I was planning on feeding an array of Greek letters to it. My current method is a somewhat workaround, but it's super janky. Any ideas?

How about getting the char code of the letter and checking against a range?

File: src/random.ts [hunk 2]
```diff
@@ -557,4 +557,79 @@ export class Random {
...
+   *
+   * @example
+   * faker.random.greekLetter() // 'θ'
+   * faker.random.greekLetter({ format: 'short' }) // 'ε'
+   * faker.random.greekLetter({ format: 'long' }) // 'lambda'
+   *
+   */
+  greekLetter(options?: { format: 'long' | 'short' }): string {
+    options.format = options.format || 'short';
+    const shortGreekLetters = [
```
Comment (ST-DDT) [inline src/random.ts:575]: Maybe move this to a file level const or separate locale file.

File: src/random.ts [hunk 3]
```diff
@@ -557,4 +557,79 @@ export class Random {
...
+   * @param options.format The format of the letter. Defaults to `'short'`.
+   *
+   * @example
+   * faker.random.greekLetter() // 'θ'
+   * faker.random.greekLetter({ format: 'short' }) // 'ε'
+   * faker.random.greekLetter({ format: 'long' }) // 'lambda'
+   *
+   */
+  greekLetter(options?: { format: 'long' | 'short' }): string {
+    options.format = options.format || 'short';
```
Comment (ST-DDT) [inline src/random.ts:574]:
```suggestion
    const format = options.format || 'short';
```

File: src/random.ts [hunk 4]
```diff
@@ -557,4 +557,79 @@ export class Random {
...
+      'tau',
+      'upsilon',
+      'phi',
+      'chi',
+      'psi',
+      'omega',
+    ];
+
+    if (options.format === 'short') {
+      return this.faker.random.arrayElement(shortGreekLetters);
```
Comment (ST-DDT) [inline src/random.ts:629]: Since this is the default case you can omit this.

File: test/random.spec.ts [hunk 5]
```diff
@@ -278,6 +278,17 @@ describe('random', () => {
...
   });
 
+  describe('greekLetter', () => {
+    it('should return a single letter', () => {
+      const greekLetter = faker.random.greekLetter();
+      // if greekLetter does not start with an English alphabet letter,
+      // check if greekLetter has only one character (Greek letters are only one character)
+      if (!/[a-zA-Z]/.test(greekLetter)) {
+        expect(greekLetter).toHaveLength(1);
+      }
```
Comment (ST-DDT) [inline test/random.spec.ts:288]: Write separate tests for long and short versions.

File: src/random.ts (hunk 2 above)
Comment (import-brain) [reply src/random.ts:575]: I don't think we would need to create any other methods dealing with Greek letters in the future, though. Wouldn't it be easier just to keep it as-is?

**Comment by Shinigami92** [comment]
I did not expect that someone suddenly start to implement this, so sorry for not inform you
We may want to have a more generic locale based function, like `letter(opts)` that returns greek letters when using `faker.locale = 'el'`
We should discuss that in discord and plan it more future-proof

**Comment by import-brain** [comment]
> I did not expect that someone suddenly start to implement this, so sorry for not inform you
> 
> We may want to have a more generic locale based function, like `letter(opts)` that returns greek letters when using `faker.locale = 'el'`
> 
> We should discuss that in discord and plan it more future-proof

That's my bad, don't worry about it. I probably should have informed you earlier.

**Comment by ST-DDT** [comment]
This PR has gone stale. The requirements have changed as well, so closing.

**Pull request closed without merging**